### PR TITLE
Enable backward mode for softmax operator

### PR DESCRIPTION
**Pull request by karthickai**
Stacked PRs:
 * __->__#528

--- --- ---

### CODE REVIEW

File: tritonbench/operators/softmax/operator.py [hunk 1]
```diff
@@ -153,8 +283,17 @@ def get_input_iter(self):
             if additional_shapes:
                 shapes.extend(additional_shapes)
 
+        requires_grad = self.mode in (Mode.BWD, Mode.FWD_BWD)
```
Comment (xuzhao9) [inline tritonbench/operators/softmax/operator.py]: We can simplify this: https://github.com/meta-pytorch/tritonbench/blob/main/tritonbench/utils/triton_op.py#L743
